models/template_helpers.go

```diff
@@ -183,7 +183,18 @@ func RemoveTemplate(q *reform.Querier, name string) error {
 		return err
 	}
 
-	err := q.Delete(&Template{Name: name})
+	rules, err := FindRules(q)
+	if err != nil {
+		return err
+	}
+
+	for _, rule := range rules {
+		if name == rule.TemplateName {
+			return errors.Errorf("failed to delete rule template, as it is being used by rule: %s", rule.ID)
+		}
+	}
```

Comment:
Better to write helper function `FindRulesByTemplateName` and not to load all rules in memory. Database can find specific rules for you.

Reply:
used reform to find rules by template name it was just two lines so didn't write a separate helper for it